File: DS&DAA-using-python/quicksort.py
#function to de partition on array
def partiton(arr,start,end):
    pivot = arr[end]
    pindex = start
    for i in range(start,end):
        if(arr[i]<=pivot):
            temp = arr[i]
            arr[i] = arr[pindex]
            arr[pindex] = temp
            pindex = pindex+1

    temp = arr[pindex]
    arr[pindex] = arr[end]
    arr[end] = temp
    return pindex

#Function to do sorting
def quicksort(arr,start,end):
    if(start < end):
        pindex = partiton(arr,start,end)
        quicksort(arr,start,pindex-1)
        quicksort(arr,pindex+1,end)


#taking array input from user
n = int(input("Enter the no of ekement in unsoterd array\n"))
arr = []
print("Enter the elements of array")
for i in range(n):
    arr.append(int(input()))

#printing unsorted array
print("Unsorted array\n",arr)

#main stetment where shorting perform
quicksort(arr,0,n-1)

#prining the sorted aaray
print("Sorted\n",arr)

File: DS&DAA-using-c/quicksort.c
#include<stdio.h>

/* Function to do partition on array*/
int partition(int arr[],int start, int end){
    int pivot = arr[end];
    int pindex = start;
    for(int i = start; i<end; i++){
        if (arr[i] <= pivot){
            int temp = arr[i];
            arr[i] = arr[pindex];
            arr[pindex] = temp;
            pindex = pindex+1;
        }
    }
    int temp = arr[pindex];
    arr[pindex] = arr[end];
    arr[end] = temp;
    return pindex;
}

/*Function to de sorting*/
void quicksort(int arr[],int start, int end){
    if(start < end){
        int pindex = partition(arr,start,end);
        quicksort(arr,start,pindex-1);
        quicksort(arr,pindex+1,end);
    }
}
void main(){
    int arr[50],n,i,j,temp,imin;
    /*Taking user input any array*/
    printf("Enter the no of element in array\n");
    scanf("%d",&n);
    printf("Enter the element of array\n");
    for(i=0;i<n;i++){
        scanf("%d",&arr[i]);       
    }
    
    /*Printing unshorted array*/
    printf("Unshorted array\n");
    for(i=0;i<n;i++){
        printf("%d\n",arr[i]);
    }
    printf("Sorting Start");

    /*main stetment where shorting perform */
    quicksort(arr,0,n);

    /*Printing shorted array*/
    printf("Shorted array\n");
    for(i=0;i<n;i++){
        printf("%d\n",arr[i]);
    }
}